daffodil-test/src/test/java/org/goodudfs/example/StringFunctions/StringFunctionsProvider.java

```diff
@@ -0,0 +1,25 @@
+package org.goodudfs.example.StringFunctions;
+
+import org.apache.daffodil.udf.*;
+
+public class StringFunctionsProvider extends UDFunctionProvider {
+	public StringFunctionsProvider() {
+		super.setFunctionClasses( new Class<?>[] { Replace.class, Compare.class } );
```

Comment:
Instead of this, a maybe safer way would be to have an ``asbtract getFunctionClasses`` method that users must implement and it returns Class array. This way, if they don't implement that function thier UDF won't compile.